lib/remap-lang/src/function.rs

```diff
@@ -117,3 +117,22 @@ pub trait Function: std::fmt::Debug {
         &[]
     }
 }
+
+pub trait CloneFunction {
```

Comment:
Can you elaborate on why this is necessary? I can't work it out..

Reply:
We can't require `Function: Clone` because then it can't be stored as a trait object.

This works around that issue by implementing a custom `clone_function()` method, which is then called if you have a `Box<dyn Function>` and call `clone` on that.